Add ID translation for define environment variable in container (#20273)
kubernetes/website: content/id/docs/tasks/inject-data-application/define-environment-variable-container.md

---

> title: Mendefinisikan Variable Lingkungan untuk sebuah Kontainer

Reviewer: ```suggestion
title: Mendefinisikan Variabel Lingkungan untuk sebuah Kontainer
```

> title: Mendefinisikan Variable-variabel Lingkungan untuk sebuah Kontainer

Author: @phanama should we use lower case in `sebuah`?

> title: Mendefinisikan Variable-variabel Lingkungan untuk sebuah Kontainer

Author: @phanama should we change `Kontainer` to `Container` in the title?

content_template: templates/task
weight: 20
---

{{% capture overview %}}

Laman ini menunjukkan bagaimana cara untuk mendefinisikan variabel lingkungan (_environment variable_) untuk sebuah Container di dalam sebuah Pod Kubernetes.

{{% /capture %}}

{{% capture prerequisites %}}

{{< include "task-tutorial-prereqs.md" >}} {{< version-check >}}

{{% /capture %}}

{{% capture steps %}}

> ## Mendefinisikan sebuah variabel lingkungan untuk sebuah kontainer

Reviewer: ```suggestion
## Mendefinisikan sebuah variabel lingkungan untuk sebuah Container
```

Reviewer: This refers to the API Object. Let's use capitalized name.

> Ketika kamu membuat sebuah Pod, kamu dapat mengatur variabel lingkungan untuk kontainer yang berjalan dalam sebuah Pod.

Author: @phanama I would suggest using `Container` only. Because I think it is not suitable to change `Container` into a plural form like Indonesian.

Untuk mengatur variabel lingkungan, sertakan bagian `env` atau `envFrom` pada berkas konfigurasi.

Dalam latihan ini, kamu membuat sebuah Pod yang menjalankan satu buah Container.

> Berkas konfigurasi untuk Pod mendefinisikan sebuah variabel lingkungan dengan nama `DEMO_GREETING` yang bernilai `"Hello from the environment"`.

Reviewer: ```suggestion
Berkas konfigurasi untuk Pod tersebut mendefinisikan sebuah variabel lingkungan dengan nama `DEMO_GREETING` yang bernilai `"Hello from the environment"`.
```

Berikut adalah berkas konfigurasi untuk Pod:

{{< codenew file="pods/inject/envars.yaml" >}}

1. Membuat sebuah Pod berdasarkan berkas konfigurasi YAML:

    ```shell
    kubectl apply -f https://k8s.io/examples/pods/inject/envars.yaml
    ```

> 2. Daftar Pod-pod yang sedang berjalan:

Author: @phanama should we use lower case in the second repeated word? Example: `Pod-pod`

Ref: https://id.wikibooks.org/wiki/Bahasa_Indonesia/Kata_ulang

    ```shell
    kubectl get pods -l purpose=demonstrate-envars
    ```

    Keluarannya mirip seperti berikut:

    ```
    NAME            READY     STATUS    RESTARTS   AGE
    envar-demo      1/1       Running   0          9s
    ```

> 3. Dapatkan _shell_ yang berjalan di Pod kamu:

Reviewer: ```suggestion
3. Dapatkan sebuah _shell_ ke Container yang berjalan di Pod kamu:
```

> 3. Dapatkan sebuah _shell_ ke Container yang berjalan di Pod kamu:

Reviewer: ```suggestion
3. Dapatkan sebuah _shell_ ke Container yang sedang berjalan di Pod kamu:
```

   ```shell
   kubectl exec -it envar-demo -- /bin/bash
   ```

> 4. Di _shell_ kamu, jalankan perintah `printenv` untuk melihat daftar variabel-variabel lingkungan.

Reviewer: ```suggestion
4. Di _shell_ kamu, jalankan perintah `printenv` untuk melihat daftar variabel lingkungan.
```

No need to use repeated form after "daftar"

> 4. Di _shell_ kamu, jalankan perintah `printenv` untuk melihat daftar variabel lingkungan.

Reviewer: ```suggestion
4. Di _shell_ kamu, jalankan perintah `printenv` untuk melihat daftar variabel lingkungannya.
```

    ```shell
    root@envar-demo:/# printenv
    ```

>     The output is similar to this:

Reviewer: ```suggestion
    Keluarannya mirip seperti ini:
```

Reviewer: Skipped translation

    ```
    NODE_VERSION=4.4.2
    EXAMPLE_SERVICE_PORT_8080_TCP_ADDR=10.3.245.237
    HOSTNAME=envar-demo
    ...
    DEMO_GREETING=Hello from the environment
    DEMO_FAREWELL=Such a sweet sorrow
    ```

> 5. Untuk keluar dari _shell_, masukkan perintah `exit`.

Reviewer: ```suggestion
5. Untuk keluar dari _shell_ tersebut, masukkan perintah `exit`.
```

{{< note >}}
Variabel-variabel lingkungan yang diatur menggunakan bagian `env` atau `envFrom` akan mengesampingkan

> variabel-variabel lingkungan yang ditentukan dalam _image_ kontainer.

Reviewer: ```suggestion
variabel-variabel lingkungan yang ditentukan di dalam _image_ kontainer.
```

{{< /note >}}

## Menggunakan variabel-variabel lingkungan di dalam konfigurasi kamu

> Variabel-variabel lingkungan yang kamu definisikan di sebuah konfigurasi Pod dapat digunakan di tempat lain dalam konfigurasi, contohnya dalam perintah-perintah dan argumen yang kamu atur dalam kontainer-kontainer Pod.

Reviewer: ```suggestion
Variabel-variabel lingkungan yang kamu definisikan di dalam sebuah konfigurasi Pod dapat digunakan di tempat lain dalam konfigurasi, contohnya dalam perintah-perintah dan argumen-argumen yang kamu atur dalam Container-Container milik Pod.
```

> Variabel-variabel lingkungan yang kamu definisikan di dalam sebuah konfigurasi Pod dapat digunakan di tempat lain dalam konfigurasi, contohnya dalam perintah-perintah dan argumen-argumen yang kamu atur dalam Container-Container milik Pod.

Reviewer: ```suggestion
Variabel-variabel lingkungan yang kamu definisikan di dalam sebuah konfigurasi Pod dapat digunakan di tempat lain dalam konfigurasi, contohnya di dalam perintah-perintah dan argumen-argumen yang kamu atur dalam Container-Container milik Pod.
```

> Pada contoh konfigurasi berikut, variabel-variabel lingkungan `GREETING`, `HONORIFIC`, dan `NAME` diatur ke `Warm greetings to`, `The Most Honorable`, dan `Kubernetes`.

Reviewer: ```suggestion
Pada contoh konfigurasi berikut, variabel-variabel lingkungan `GREETING`, `HONORIFIC`, dan `NAME` disetel masing-masing menjadi `Warm greetings to`, `The Most Honorable`, dan `Kubernetes`.
```

> Variabel-variabel lingkungan tersebut kemudian digunakan dalam argumen CLI yang diteruskan ke kontainer `env-print-demo`.

Reviewer: ```suggestion
Variabel-variabel lingkungan tersebut kemudian digunakan dalam argumen CLI yang diteruskan ke Container `env-print-demo`.
```

```yaml
apiVersion: v1
kind: Pod
metadata:
  name: print-greeting
spec:
  containers:
  - name: env-print-demo
    image: bash
    env:
    - name: GREETING
      value: "Warm greetings to"
    - name: HONORIFIC
      value: "The Most Honorable"
    - name: NAME
      value: "Kubernetes"
    command: ["echo"]
    args: ["$(GREETING) $(HONORIFIC) $(NAME)"]
```

> Setelah dibuat, perintah `echo Warm greetings to The Most Honorable Kubernetes` dijalankan di kontainer.

Reviewer: ```suggestion
Setelah dibuat, perintah `echo Warm greetings to The Most Honorable Kubernetes` dijalankan di Container tersebut.
```
